csharp/extractor/Semmle.Extraction/Entity.cs

```diff
@@ -96,48 +98,36 @@ public interface ICachedEntityFactory<Initializer, Entity> where Entity : ICache
     public static class ICachedEntityFactoryExtensions
     {
         public static Entity CreateEntity<Entity, T1, T2>(this ICachedEntityFactory<(T1, T2), Entity> factory, Context cx, T1 t1, T2 t2)
-            where Entity : ICachedEntity
-        {
-            return factory.CreateEntity(cx, (t1, t2));
-        }
+            where Entity : ICachedEntity => factory.CreateEntity2(cx, (t1, t2));
```

Comment:
Why not `factory.CreateEntity` -- `(t1, t2)` is never `null`? Same below

Reply:
Following on from the comment above - you want to use `Tuple.Item1` as the key (the other columns may be different), which is the behaviour of `CreateEntity2`.